QuizApp/FinishedQuiz/Presenter/QuizFinishedPresenter.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  QuizFinishedPresenter.swift
+//  QuizApp
+//
+//  Created by Luka Cicak on 15.05.2021..
+//
+
+import Foundation
+import UIKit
+
+class QuizFinishedPresenter {
```

Comment:
To avoid sending your ViewController to every single function in presenter you can use the delegate pattern and then your presenter would look something like this:
```swift
class QuizFinishedPresenter {

    weak var delegate: QuizFinishedPresenterDelegate?

    func resultsUploaded() {
        DispatchQueue.main.async {
            delegate.showSuccessPopup()
        }
    }

    func sendResults() {
        // setup everything
      
        networkService.executeUrlRequest(request) { [weak self] result in 
            guard let self = self else { return }
  
            switch result {
            case .success(let value):
                guard value.error == nil else { return }

                self.resultsUploaded()
            ...
            }
        }
    }
}
```
and here's the look of the view controller:
```swift
class QuizFinishedViewController: UIViewController {

    private var presenter: QuizFinishedPresenter!

    init(presenter: QuizFinishedPresenter) {
        self.presenter = presenter

        presenter.delegate = self
    }

...
}

extension QuizFinishedViewController: QuizFinishedPresenterDelegate {

    func showSuccessPopup() {
        // do UI stuff
    }

}
```